File: developer/Samples/NET 4.6/WF_WCF_Samples/WCF/Basic/Binding/ByteStreamEncoder/CS/ByteStreamHttpBinding/HostNameComparisonModeHelper.cs
//----------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//----------------------------------------------------------------

using System.ComponentModel;
using System.ServiceModel;

namespace Microsoft.Samples.ByteStreamEncoder
{

    static class HostNameComparisonModeHelper
    {
        public static void Validate(HostNameComparisonMode value)
        {
            if (!IsDefined(value))
            {
                throw(new InvalidEnumArgumentException("value", (int)value,
                    typeof(HostNameComparisonMode)));
            }
        }
        internal static bool IsDefined(HostNameComparisonMode value)
        {
            return
                value == HostNameComparisonMode.StrongWildcard
                || value == HostNameComparisonMode.Exact
                || value == HostNameComparisonMode.WeakWildcard;
        }
    }
}

File: developer/Samples/NET 4.6/WF_WCF_Samples/WCF/Basic/Binding/ByteStreamEncoder/VB/ByteStreamHttpBinding/HostNameComparisonModeHelper.vb
'----------------------------------------------------------------
' Copyright (c) Microsoft Corporation.  All rights reserved.
'----------------------------------------------------------------

Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports System.ServiceModel

Namespace Microsoft.Samples.ByteStreamEncoder

    Friend NotInheritable Class HostNameComparisonModeHelper

        Private Sub New()
        End Sub

        Public Shared Sub Validate(ByVal value As HostNameComparisonMode)
            If (Not IsDefined(value)) Then
                Throw (New InvalidEnumArgumentException("value", CInt(Fix(value)), GetType(HostNameComparisonMode)))
            End If
        End Sub

        Friend Shared Function IsDefined(ByVal value As HostNameComparisonMode) As Boolean
            Return value = HostNameComparisonMode.StrongWildcard OrElse value = HostNameComparisonMode.Exact OrElse value = HostNameComparisonMode.WeakWildcard
        End Function
    End Class

End Namespace
